server/src/services/item.service.ts

```diff
@@ -61,6 +77,25 @@ class ItemService {
       limit,
     });
   }
+
+  async autoCompleteKeyword(keyword: string): Promise<string[]> {
+    const result = await this._searchToElasticSearch(keyword);
+    const titles = result.hits.hits.map((hit) => {
+      const { title } = hit._source as Item;
+      return title;
+    });
+
+    return titles;
+  }
+  async searchItem(keyword: string): Promise<Item[]> {
+    const result = await this._searchToElasticSearch(keyword);
+    const items = result.hits.hits.map((hit) => {
+      const item = hit._source as Item;
+      return item;
+    });
```

Comment:
```suggestion
    const items = result.hits.hits.map(({ _source }) => _source) as Item[];
```

이렇게도 표현할 수 있을 것네요.
```suggestion
    const items = result.hits.hits.map((hit) => hit._source as Item);
```
아님 이렇게도?